### PR TITLE
Certificate facts

### PR DESCRIPTION
**Pull request by minorOffense**
#### Pull Request (PR) description
Add facts about installed certificates so puppet code can make decisions based on the status of certificates.

#### This Pull Request (PR) fixes the following issues
Came from a discussion in https://github.com/voxpupuli/puppet-letsencrypt/issues/66

We've actually had this code around for a while and have been using it in production for over a year. I forgot to open the pull request earlier...

Credit should be given as well to @cwells who contributed the code for this.

### CODE REVIEW

**Comment by minorOffense** [comment]
Ok, all refactored and still working. Ready for review

**Comment by dhoppe** [comment]
@minorOffense Thank you very much for your contribution. Do you mind updating the documentation and adding some tests?

**Comment by minorOffense** [comment]
I don’t mind updating the docs but I’m not sure how to write tests for this since it scans the contents of a certificate. Does your test suite make cert requests already? Or is there a dummy cert that’s added to the test run?

**Comment by dhoppe** [comment]
@minorOffense Looks like we only have unit tests, but do not do any acceptance tests. https://github.com/voxpupuli/puppet-letsencrypt/blob/master/spec/defines/letsencrypt_certonly_spec.rb

**Comment by Dan33l** [comment]
@dhoppe  some basics acceptance tests are done. But we do not get certificate during tests.

**Comment by dhoppe** [comment]
@Dan33l Thx for the feedback. @minorOffense Please update the docs and were are good to go. ;)

**Comment by minorOffense** [comment]
Sorry I was busy last week. I'll write up some docs and show a sample output from facter in a few different cases (invalid cert, valid, multiple domain cert). Should be ready for the end of the week.

**Comment by minorOffense** [comment]
Sorry I forgot to remove the WIP. But yeah I added docs to it. Let me know if that's sufficient.

File: lib/facter/letsencrypt_directory.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'openssl'
+require 'pathname'
+
+Facter.add(:letsencrypt_directory) do
+  setcode do
```
Comment (Dan33l) [inline lib/facter/letsencrypt_directory.rb]: since this is not expected to work with windows node, the fact should be confined to Linux kernel.

**Comment by minorOffense** [comment]
> I am not a ruby dev but i added an in line comment.
> And please squash your commits.

I'll add a note in the fact. For the squash I figured whoever merges it just uses the "Squash and merge" button in github. But I can "pre-squash" things.

File: lib/facter/letsencrypt_directory.rb (hunk 1 above)
Comment (ekohl) [reply lib/facter/letsencrypt_directory.rb]: How about BSD kernels? I'd expect it to work there as well.

Comment (Dan33l) [reply lib/facter/letsencrypt_directory.rb]: Ah yes ! The module have a huge list of compatibles OSes.
This should work:
`confine :kernel => [:Linux, :FreeBSD, :OpenBSD]`

What i wanted to escape is to run the code of fact on windows.

Comment (minorOffense) [reply lib/facter/letsencrypt_directory.rb]: Added the confine. And matched up to whatever rubocop wanted.

Comment (minorOffense) [reply lib/facter/letsencrypt_directory.rb]: Whoops I did not squash that right...

**Comment by minorOffense** [comment]
Ah I really messed up this branch. Ran one wrong git command and I've been digging a deeper hole ever since. I'm just going to reopen a fresh PR on a branch with everything correctly set.

**Comment by minorOffense** [comment]
Ok I've fixed it all up and it's in https://github.com/voxpupuli/puppet-letsencrypt/pull/187

**Pull request closed without merging**